sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2037 @@
+///
+/// ## Arguments
+///
+/// * `font_path` - File path to the font file to read reference from.
+///
+/// ## Returns
+/// If a reference is available, it will be returned.
+#[allow(dead_code)]
+fn read_reference_from_font(font_path: &Path) -> Result<Option<String>> {
+    // open the font source
+    let mut font_stream = open_bufreader_for_file(font_path)?;
+    read_reference_to_stream(&mut font_stream)
+}
+
+/// Reads the C2PA manifest store reference from the stream.
+///
+/// ## Arguments
+///
+/// * `source` - Source font stream to read reference from.
+///
+/// ## Returns
+/// If a reference is available, it will be returned.
+#[allow(dead_code)]
+fn read_reference_to_stream<TSource>(source: &mut TSource) -> Result<Option<String>>
+where
+    TSource: Read + Seek + ?Sized,
+{
+    match read_c2pa_from_stream(source) {
+        Ok(c2pa_data) => Ok(c2pa_data.activeManifestUri.to_owned()),
+        Err(Error::JumbfNotFound) => Ok(None),
+        Err(_) => Err(Error::DeserializationError),
+    }
+}
+
+/// Remove the `C2PA` font table from the font file.
+///
+/// ## Arguments
+///
+/// * `font_path` - path to the font file to remove C2PA from
+fn remove_c2pa_from_font(font_path: &Path) -> Result<()> {
+    process_file_with_streams(font_path, move |input_stream, temp_file| {
+        // Remove the C2PA manifest store from the stream
+        remove_c2pa_from_stream(input_stream, temp_file.get_mut_file())
+    })
+}
+
+/// Remove the `C2PA` font table from the font data stream, writing to the
+/// destination.
+///
+/// ## Arguments
+///
+/// * `source` - Source data stream containing font data
+/// * `destination` - Destination data stream to write new font data with the
+///                   C2PA table removed
+fn remove_c2pa_from_stream<TSource, TDest>(
+    source: &mut TSource,
+    destination: &mut TDest,
+) -> Result<()>
+where
+    TSource: Read + Seek + ?Sized,
+    TDest: Write + ?Sized,
+{
+    source.rewind()?;
+    // Load the font from the stream
+    let mut font = Font::read(source).map_err(|_| Error::FontLoadError)?;
+    // Remove the table from the collection
+    font.tables.remove(&C2PA_TABLE_TAG);
+    // And write it to the destination stream
+    font.write(destination).map_err(|_| Error::FontSaveError)?;
+
+    Ok(())
+}
+
+/// Removes the reference to the active manifest from the source stream, writing
+/// to the destination.
+///
+/// ## Arguments
+///
+/// * `source` - Source data stream containing font data
+/// * `destination` - Destination data stream to write new font data with the
+///                   active manifest reference removed
+///
+/// ## Returns
+///
+/// The active manifest URI reference that was removed, if there was one
+#[allow(dead_code)]
+fn remove_reference_from_stream<TSource, TDest>(
+    source: &mut TSource,
+    destination: &mut TDest,
+) -> Result<Option<String>>
+where
+    TSource: Read + Seek + ?Sized,
+    TDest: Write + ?Sized,
+{
```

Comment:
@brogdonm, should this function commence with a `source.rewind()?;`, like `add_refererence_to_stream` does?

Reply:
Theoretically, yes. But the SDK doesn't utilize the path, so it wouldn't matter. Maybe we remove the `remove_ref*` function?

Reply:
1. Rewinding the stream = belt + suspenders.
1. Move to table discussion for elswhen.